/*
    GSP for the TF blockchain game
    Copyright (C) 2020  Autonomous Worlds Ltd

    This program is free software: you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
    the Free Software Foundation, either version 3 of the License, or
    (at your option) any later version.

    This program is distributed in the hope that it will be useful,
    but WITHOUT ANY WARRANTY; without even the implied warranty of
    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
    GNU General Public License for more details.

    You should have received a copy of the GNU General Public License
    along with this program.  If not, see <https://www.gnu.org/licenses/>.
*/

syntax = "proto2";
option cc_enable_arenas = true;

package pxd.proto;

message TournamentBlueprint
{
  /*Id*/
  optional string AuthoredId = 1;
    
  /*n*/
  optional string Name = 2;
  
  /*d*/
  optional uint32 Duration = 3;  
  
  /*tc*/
  optional uint32 TeamCount = 4;

  /*ts*/
  optional uint32 TeamSize = 5;

  /*mis*/
  optional uint32 MinSweetness = 6; 
  
  /*mas*/
  optional uint32 MaxSweetness = 7;

  /*mrq*/
  optional uint32 MaxRewardQuality = 8; 

  /*brt*/
  optional string BaseRewardsTableId = 9;  

  /*brc*/
  optional uint32 BaseRollCount = 10; 

  /*wrt*/
  optional string WinnerRewardsTableId = 11;

  /*wrc*/
  optional uint32 WinnerRollCount = 12;    
  
  /*jc*/
  optional uint32 JoinCost = 13;    
}
